test/shopify_app/login_protection_test.rb

```diff
@@ -86,13 +90,21 @@ class LoginProtectionTest < ActionController::TestCase
     end
   end
 
+  test '#shopify_session when rescuing from unauthorized access, redirects to the login url' do
+    with_application_test_routes do
+      get :raise_unauthorized, shop: 'foobar'
+      assert_redirected_to @controller.send(:main_or_engine_login_url, shop: 'foobar')
+    end
+  end
```

Comment:
I just double checked what the behaviour is with an empty shop param and it should work: https://github.com/Shopify/shopify_app/blob/master/lib/shopify_app/sessions_concern.rb#L6. We have tests for this in the session controller test